### PR TITLE
Automatically open a browser window when Dev Server starts

### LINKED ISSUE #92: `carton dev` should open a browser window when server starts
This would be more consistent with what `react-scripts` is doing and I think would be a nice usability improvement.

### PR DESCRIPTION
**Pull request by yonihemi**
Resolves #92.
While there [a lot](https://github.com/facebook/create-react-app/blob/master/packages/react-dev-utils/openBrowser.js) [more](https://github.com/sindresorhus/open) options possible to cram here I think this is a decent start. Perhaps an opportunity for a cross-platform Swift launcher µ framework?

### CODE REVIEW

**Review by MaxDesiatov — APPROVED** [review]
Yes! This is wonderful, thank you so much 🙏

**Comment by MaxDesiatov** [comment]
Thanks again for your contributions, I've sent you an invitation to the @swiftwasm/carton-team team if you're interested in joining, feel free to merge it yourself after you accept it, or wait for a few more approvals if you want to get something close to a team consensus (not required, but desirable, unless we rush some important pre-release PRs 🙂)

**Review by carson-katri — APPROVED** [review]

**Review by RayZhao1998 — APPROVED** [review]

**Pull request merged**
